tool/teleport/common/wait.go

```diff
@@ -124,44 +126,57 @@ func waitNoResolve(ctx context.Context, domain string, period, timeout time.Dura
 			return trace.Wrap(err)
 
 		case <-periodic.Next():
-			exit, err = checkDomainNoResolve(domain)
+			exit, err = checkDomainNoResolve(ctx, domain, log)
 			if err != nil {
 				return trace.Wrap(err)
 			}
 		}
 	}
 
-	log.Info("no endpoints found, exiting with success code")
+	log.InfoContext(ctx, "no endpoints found, exiting with success code")
 	return nil
 }
 
-func checkDomainNoResolve(domainName string) (exit bool, err error) {
-	endpoints, err := countEndpoints(domainName)
+func checkDomainNoResolve(ctx context.Context, domainName string, log *slog.Logger) (exit bool, err error) {
+	endpoints, err := resolveEndpoints(domainName)
 	if err != nil {
 		var dnsErr *net.DNSError
 		if !errors.As(trace.Unwrap(err), &dnsErr) {
-			log.Errorf("unexpected error when resolving domain %s : %s", domainName, err)
+			log.ErrorContext(ctx, "unexpected error when resolving domain", "error", err)
 			return false, trace.Wrap(err)
 		}
+		dnsErrorDetails := slog.Group("dns_error",
+			"name", dnsErr.Name,
+			"server", dnsErr.Server,
+			"is_timeout", dnsErr.IsTimeout,
+			"is_temporary", dnsErr.IsTemporary,
+			"is_not_found", dnsErr.IsNotFound,
+			// Logging the error type can help understanding where the error comes from
+			"wrapped_error_type", reflect.TypeOf(dnsErr.Unwrap()).String(),
```

Comment:
I know this is sketchy, but I faced obscure DNS issues with some kube stack and that's my last resort before firing `tcpdump`

Reply:
Just curios why not just: 
```suggestion
			// Logging the error type can help understanding where the error comes from
			"wrapped_error_type", fmt.Sprintf("%T", dnsErr.Unwrap()),
```